src/python/pants/backend/python/tasks/resolve_requirements.py

```diff
@@ -28,5 +32,19 @@ def execute(self):
     if not self.context.targets(lambda t: is_python_target(t) or has_python_requirements(t)):
       return
     interpreter = self.context.products.get_data(PythonInterpreter)
-    pex = self.resolve_requirements(interpreter, self.context.targets(has_python_requirements))
-    self.context.products.register_data(self.REQUIREMENTS_PEX, pex)
+
+    pre_requirement_targets = []
```

Comment:
So, if it is the case that the ordering of requirements is preserved in some way by `self.get_targets` (perhaps due to use of DFS or BFS such that the spec roots are first in the list), it feels like it would be possible to force a requirement to load first by declaring a dependency on the problematic requirement "very close to" your binary or test.

I'm fine with this as a workaround for requirements ordering issues, but on the JVM side (and with dependencies lists in general) we preserve ordering for this reason.

Reply:
I have a vague hunch it's because we provide `setuptools` in at least one of the pexes we combine for the pytest pex, and that version of setuptools conflicts with the one that `google-cloud-bigquery` expects, but doesn't declare (which pex now warns about), and that causes the `google-cloud-bigquery` package to read from a different setuptools, and this causes it to have the failure mentioned in the PR description (just a hunch). Not sure whether/how to act on this.

Reply:
...should an explicit modern dependency be added to `google-cloud-bigquery` then, maybe?